### PR TITLE
accounts/abi: add ErrorById

### PR DESCRIPTION
**Pull request by chawin-a**
### Changes
- Modify ID type from `common.Hash` to `[]byte` in Error struct
- Add ErrorById

### CODE REVIEW

File: accounts/abi/abi.go [hunk 1]
```diff
@@ -222,6 +222,20 @@ func (abi *ABI) EventByID(topic common.Hash) (*Event, error) {
...
 }
 
+// ErrorById looks up an error by the 4-byte id,
+// returns nil if none found.
+func (abi *ABI) ErrorById(sigdata []byte) (*Error, error) {
+	if len(sigdata) < 4 {
+		return nil, fmt.Errorf("data too short (%d bytes) for abi error lookup", len(sigdata))
+	}
+	for _, errABI := range abi.Errors {
+		if bytes.Equal(errABI.ID, sigdata[:4]) {
```
Comment (fjl) [inline accounts/abi/abi.go:232]: I wonder why you changed the `ID` field to `[]byte`. Might be better to:

- change the type of the `sigdata` parameter to `[4]byte`
- do the check like bytes.Equal(errABI.ID[:4], sigdata[:])`

**Comment by deep709** [comment]
Thanks all

**Comment by fjl** [comment]
Please apply the requested changes!

**Comment by chawin-a** [comment]
Done please review the changes

File: accounts/abi/error.go [hunk 2]
```diff
@@ -38,7 +37,7 @@ type Error struct {
 
 	// ID returns the canonical representation of the error's signature used by the
 	// abi definition to identify event names and types.
-	ID common.Hash
+	ID [4]byte
```
Comment (fjl) [inline accounts/abi/error.go:40]: You don't need to change the `ID` to make the function `ErrorByID` work.

File: accounts/abi/abi.go [hunk 3]
```diff
@@ -222,6 +222,17 @@ func (abi *ABI) EventByID(topic common.Hash) (*Event, error) {
 	return nil, fmt.Errorf("no event with id: %#x", topic.Hex())
 }
 
+// ErrorById looks up an error by the 4-byte id,
+// returns nil if none found.
+func (abi *ABI) ErrorById(sigdata [4]byte) (*Error, error) {
```
Comment (fjl) [inline accounts/abi/abi.go:227]: Please rename this method to `ErrorByID`

**Review by fjl — CHANGES_REQUESTED** [review]
Sorry some more changes needed

**Comment by chawin-a** [comment]
Please review the changes again.

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM

**Pull request merged**